### PR TITLE
Change pT5 rz chi2 cut definition into helix approximation (rebase PR40)

### PR DESCRIPTION
**Pull request by ariostas**
This is a rebase of PR #40. Since it changes the performance I made it as its own PR so that we can compare the plots with the original PR.

### CODE REVIEW

File: RecoTracker/LSTCore/src/alpaka/PixelTriplet.h [hunk 1]
```diff
@@ -641,10 +642,9 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE::lst {
         float diffz1 = alpaka::math::abs(acc, solz1 - zsi) * 100;
         float diffz2 = alpaka::math::abs(acc, solz2 - zsi) * 100;
         diffz = alpaka::math::min(acc, diffz1, diffz2);
+        residual = diffz;
```
Comment (YonsiG) [inline RecoTracker/LSTCore/src/alpaka/PixelTriplet.h:645]: It is a bit weird that this line and the Line 628 are added into this PR, it seems pure pT3 rzchi2 work.

Comment (ariostas) [reply RecoTracker/LSTCore/src/alpaka/PixelTriplet.h:645]: Thank you, @YonsiG. Seems like I might have messed something up. I'll fix it.

Comment (ariostas) [reply RecoTracker/LSTCore/src/alpaka/PixelTriplet.h]: Actually, if you look at #40, those changes were made for pT3 rzchi2. Either way, those changes shouldn't affect anything.

Comment (YonsiG) [reply RecoTracker/LSTCore/src/alpaka/PixelTriplet.h]: Yes, those are for pT3 rzchi2. The order doesn't affect much as you said.

**Pull request merged**